compiler/importer.nim

```diff
@@ -68,6 +68,22 @@ proc rawImportSymbol(c: PContext, s: PSym) =
     if s.kind == skConverter: addConverter(c, s)
     if hasPattern(s): addPattern(c, s)
 
+proc maybeImportForType(c: PContext, s: PSym, t: PType) =
+  ## Imports a symbol only if it has a strict relation with a type:
+  ##
+  ##  - Proc refers to the type on any of its input or output values.
+  ##
+  ## TODO: generics are not matched currently.
+  ##
+  case s.kind
+  of skProcKinds:
+    for tt in s.typ.sons.items:
```

Comment:
Question: shall I avoid using iterators in the compiler code? maybe they are not as efficient as `countup` loops?

Reply:
No need to avoid them, they are zero-cost.